src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -372,10 +372,20 @@ internal bool CheckValueKind(SyntaxNode node, BoundExpression expr, BindValueKin
 
                 case BoundKind.ArrayAccess:
                 case BoundKind.PointerIndirectionOperator:
-                case BoundKind.PointerElementAccess:
                     // array elements and pointer dereferencing are readwrite variables
                     return true;
 
+                case BoundKind.PointerElementAccess:
+                    {
+                        var receiver = ((BoundPointerElementAccess)expr).Expression;
+                        if (receiver is BoundFieldAccess fieldAccess && fieldAccess.FieldSymbol.IsFixed)
+                        {
+                            return CheckValueKind(node, fieldAccess.ReceiverOpt, valueKind, checkingReceiver: true, diagnostics);
```

Comment:
>return CheckValueKind(node, fieldAccess.ReceiverOpt, valueKind, checkingReceiver: true, diagnostics); [](start = 28, length = 101)

Do we have a test that demonstrates that this new logic makes a difference?

Reply:
Yes. For example `CS1708ERR_FixedNeedsLvalue`.